src/main/java/duckie/Storage.java

```diff
@@ -0,0 +1,148 @@
+                } else {
+                    if (duckieFile.createNewFile()) {
+                        System.out.println("\t" + "Memory File created successfully!");
+                    } else {
+                        throw new DuckieException("Quack! Unable to create file!");
+                    }
+                }
+            }
+            return duckieFile;
+        } catch (IOException e) {
+            throw new DuckieFileErrorException();
+        }
+    }
+
+    /**
+     * Load the duckie file and generate the ArrayList containing the tasks
+     * @return ArrayList containing all the saved tasks
+     * @throws DuckieException
+     */
+    public ArrayList<Task> load() throws DuckieException {
+        File duckieFile = openFile();
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(duckieFile);
+            while (sc.hasNextLine()) {
+                String task = sc.nextLine();
+                String[] taskBreakdown = task.split("\\|");
+
+                String type = taskBreakdown[0].strip();
+                String isDone = taskBreakdown[1].strip();
+                String description = taskBreakdown[2].strip();
+                switch (type) {
+                case "T":
+                    Todo taskToDo = new Todo(description);
+                    if (isDone.equals("1")) {
+                        taskToDo.markDone();
+                    }
+                    tasks.add(taskToDo);
+                    break;
+                case "D":
+                    String date = taskBreakdown[3].strip();
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("E, MMM dd yyyy");
+                    LocalDate d1 = LocalDate.parse(date, formatter);
+                    Deadline taskD = new Deadline(description, d1);
+                    if (isDone.equals("1")) {
+                        taskD.markDone();
+                    }
+                    tasks.add(taskD);
+                    break;
+                case "E":
+                    String dateTime = taskBreakdown[3].strip();
+                    DateTimeFormatter formatter2 = DateTimeFormatter.ofPattern("E, MMM dd yyyy hh:mm a");
+                    LocalDateTime d2 = LocalDateTime.parse(dateTime, formatter2);
+                    Event taskE = new Event(description, d2);
+                    if (isDone.equals("1")) {
+                        taskE.markDone();
+                    }
+                    tasks.add(taskE);
+                    break;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            throw new DuckieException("Quack! Duckie cannot find your File!");
+        }
+
+        if (tasks.size() == 0) {
+            Ui.displayNoListReply();
+        } else {
+            Ui.displayListReply(tasks);
+            Ui.showLine();
+        }
+        return tasks;
+    }
+
+    /**
+     * Update the current tasks in the TaskList to the duckie file
```

Comment:
Should there be an 'empty line between description and parameter section'?

Reply:
I am sorry but I noticed that you have copy pasted this same comment like multiple times so far. Mentioning it once or twice already makes the point I feel.